scene/audio/audio_stream_player.cpp

```diff
@@ -106,57 +105,80 @@ void AudioStreamPlayer::set_pitch_scale(float p_pitch_scale) {
 	ERR_FAIL_COND(p_pitch_scale <= 0.0);
 	pitch_scale = p_pitch_scale;
 
-	AudioServer::get_singleton()->set_playback_pitch_scale(stream_playback, pitch_scale);
+	for (Ref<AudioStreamPlayback> &playback : stream_playbacks) {
+		AudioServer::get_singleton()->set_playback_pitch_scale(playback, pitch_scale);
+	}
 }
 
 float AudioStreamPlayer::get_pitch_scale() const {
 	return pitch_scale;
 }
 
+void AudioStreamPlayer::set_max_polyphony(int p_max_polyphony) {
+	if (p_max_polyphony > 0) {
+		max_polyphony = p_max_polyphony;
+	}
+}
+
+int AudioStreamPlayer::get_max_polyphony() const {
+	return max_polyphony;
+}
+
 void AudioStreamPlayer::play(float p_from_pos) {
-	stop();
-	if (stream.is_valid()) {
-		stream_playback = stream->instance_playback();
+	if (stream.is_null()) {
+		return;
 	}
-	if (stream_playback.is_valid()) {
-		AudioServer::get_singleton()->start_playback_stream(stream_playback, bus, _get_volume_vector(), p_from_pos);
-		active.set();
+	Ref<AudioStreamPlayback> stream_playback = stream->instance_playback();
+	ERR_FAIL_COND_MSG(stream_playback.is_null(), "Failed to instantiate playback.");
+
+	AudioServer::get_singleton()->start_playback_stream(stream_playback, bus, _get_volume_vector(), p_from_pos);
+	stream_playbacks.push_back(stream_playback);
+	active.set();
+	set_process_internal(true);
+
+	if (stream_playbacks.size() > max_polyphony) {
```

Comment:
Since you are at it, I suggest adding an extra method `AudioStream::is_polyphonic`, which should return true by default, but for some like the microphone it should return false. This should be checked here also to ensure no more than one can be played at the same time, else all it will result is in the audio duplicating volume.

Reply:
Added. I know you mentioned this earlier in rocket.chat but I must have forgotten.